### DISCUSSION TITLE
Add support to DMG's for drag to install to Application folder on MacOS

### DISCUSSION THREAD

**Discussion by ntindle** [marcelotduarte/cx_Freeze#2438] [Ideas]
Tools like [`dmgbuild` (python)](https://github.com/dmgbuild/dmgbuild) and [`appdmg` (node)](https://www.npmjs.com/package/appdmg) for MacOS allow the user to configure the layout of the `.dmg` drive and add things like this

![img of a visually configured dmg](https://raw.githubusercontent.com/LinusU/node-appdmg/bb0f693a74341fda7934b42fd0adcb37022f6e05/help/help.png)

Would you be interested in having that feature for cx_Freeze, and if so, how would you expect it to be integrated? Reimplementation of the existing package (dmgbuild) or using it?

The work to add it on doesn't seem to be that large and with guidance I would be willing to put up a pr

**Comment by marcelotduarte** [comment]
I'm not a macOS user, but I knew about dmgbuild and [create-dmg](https://github.com/create-dmg/create-dmg), which seem more complete, but more difficult to integrate.
And answering your question, I am happy with your willingness to contribute and I believe that using dmgbuild as a dependency is the best option.
The implementation of bdist_dmg in this case, should be moved from [here](https://github.com/marcelotduarte/cx_Freeze/blob/main/cx_Freeze/command/bdist_mac.py#L27) to a new module.
There a test [here](https://github.com/marcelotduarte/cx_Freeze/blob/main/tests/test_command_bdist_mac.py#L46) that can be moved to new module and enhanced to test the new options.

**Reply by ntindle** [reply]
#2442
